### PR TITLE
bradl3yC 3883 Add edit address link

### PR DESCRIPTION
**Pull request by bradl3yC**
## Description
Add Edit Address link after user entered address

## Testing done
Unit tests modified

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by erikphansen — APPROVED** [review]

File: src/platform/user/profile/vet360/containers/AddressValidationModal.jsx [hunk 1]
```diff
@@ -104,6 +109,12 @@ class AddressValidationModal extends React.Component {
             {addressLine2 && <span>{addressLine2}</span>}
             {addressLine3 && <span>{addressLine3}</span>}
             <span>{` ${city}, ${stateCode} ${zipCode}`}</span>
+            {isAddressFromUser &&
+              !addressValidationError && (
+                <a onClick={() => this.props.openModal(addressValidationType)}>
```
[the remark this answers is not in the document]
Comment (bradl3yC) [reply src/platform/user/profile/vet360/containers/AddressValidationModal.jsx]: This results in an 'already defined in upper scope' lint error

[the remark this answers is not in the document]
Comment (erikphansen) [reply src/platform/user/profile/vet360/containers/AddressValidationModal.jsx]: Ah right. sometimes when I run into that I just do exactly what you did here. and sometimes I rename the conflicting import, so in this case I'd do `import { openModal as openModalAction, createTransaction } from '../actions';` I started to do the latter just so I could pull everything off of the props at the top of function I'm working in. But you can totally leave this.

**Comment by bradl3yC** [comment]
@department-of-veterans-affairs/frontend-review-group ready for review.  This is just to add `Edit Address` to the FE when necessary.

File: src/platform/user/profile/vet360/containers/AddressValidationModal.jsx [hunk 2]
```diff
@@ -83,7 +87,8 @@ class AddressValidationModal extends React.Component {
       zipCode,
     } = address;
 
-    const isOptionDisabled = id === 'userEntered' && !validationKey;
+    const isAddressFromUser = id === 'userEntered';
```
Comment (ncksllvn) [inline src/platform/user/profile/vet360/containers/AddressValidationModal.jsx]: `'userEntered'` could be a good candidate for a constant

File: src/platform/user/profile/vet360/containers/AddressValidationModal.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (ncksllvn) [reply src/platform/user/profile/vet360/containers/AddressValidationModal.jsx]: Yeah that lint error is a little annoying when working with containers

**Review by ncksllvn — APPROVED** [review]

**Review by U-DON — APPROVED** [review]

**Pull request merged**